server/server.go

```diff
@@ -75,37 +89,45 @@ func LoginUser(w http.ResponseWriter, res *http.Request) {
 		return
 	}
 
-	//set up JWT ====================
-	expirationTime := time.Now().Add(time.Minute * 5)
-
-	claims := &data.Claims{
-		Username: returningUser.Username,
-		StandardClaims: jwt.StandardClaims{
-			ExpiresAt: expirationTime.Unix(),
-		},
+	//Create JWT, store it in jwtToken var
+	jwtToken, signingErr := createJWT(returningUser.Username)
+	if signingErr != nil {
+		respondWithError(w, "Internal Server Error", http.StatusInternalServerError)
 	}
 
-	jwtKey := []byte("secret")
-	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
-	tokenString, err := token.SignedString(jwtKey)
+	sendJWT(w, jwtToken)
 
-	if err != nil {
-		respondWithError(w, "Internal Server Error", http.StatusInternalServerError)
-		return
-	}
+}
+
+// Handle /implicit_login
+func handleImplicitLogin(w http.ResponseWriter, res *http.Request) {
+	headers := res.Header
+	authHeader := headers.Get("Authentication")
+	fmt.Printf("AuthenticationHeader = %v\n", authHeader)
 
-	//temporarily print tokenString, to remove "unused variable" error
-	fmt.Println(tokenString)
 }
 
-// Handle Error Response =====================================================
+//Utility Functions ========================================================================================
+
+// Handle Error Response
 func respondWithError(w http.ResponseWriter, msg string, code int) {
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(code)
 	json.NewEncoder(w).Encode(data.ApiError{ErrorMessage: msg})
 
 }
 
+// Encode JWT Token into JSON
+func sendJWT(w http.ResponseWriter, jwt string) {
+	json.NewEncoder(w).Encode(data.TokenMessage{Token: jwt})
```

Comment:
It's a good practice to explicitly set the content-type header to "application/json" when returning json. You could do this here. (Line 114 shows how)

Reply:
https://scratched-ethernet-1a9.notion.site/Documentation-ddd34b987fe245e08dc405983878e8ee